x/twap/strategy.go

```diff
@@ -0,0 +1,43 @@
+package twap
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/v13/x/twap/types"
+)
+
+// twapStrategy is an interface for computing TWAPs.
+// We have two strategies implementing the interface - arithmetic and geometric.
+// We expose a common TWAP API to reduce duplication and avoid complexity.
+type twapStrategy interface {
+	// getTwapToNow calculates the TWAP with endRecord as currentBlocktime.
+	getTwapToNow(
+		ctx sdk.Context,
+		poolId uint64,
+		baseAssetDenom string,
+		quoteAssetDenom string,
+		startTime time.Time) (sdk.Dec, error)
+	// computeTwap calculates the TWAP with specific startRecord and endRecord.
+	computeTwap(startRecord types.TwapRecord, endRecord types.TwapRecord, quoteAsset string) (sdk.Dec, error)
```

Comment:
nit: Should we unify these names: either both be called "get" or both called "compute"?

Reply:
also. Is there a reason to have `getTwapToNow` be part of the strategy?

Reply:
i think unifying makes sense, i'm up for naming it `getTwapToNow` and `getTwap`. @p0mvn lmk what you think of this

For `getTwapToNow` I was calling the keeper method directly [here](https://github.com/osmosis-labs/osmosis/pull/3529/files#diff-4d165167a6ae9c6e7d42cffc254bb4a98eb339ccf26be3a032a40ca2729a5ce8R87) but i changed it to use strategy.go route